test/test_calendar.py

```diff
@@ -150,3 +156,137 @@ def test_parse_date_offset_no_offset_magnitude():
 
     expected_error = "No numerical value given for offset: YS"
     assert str(exc_info.value) == expected_error
+
+
+@pytest.mark.parametrize(
+        "start_date, end_date, caltype_int, expected",
+        [
+            (
+                datetime.datetime(year=4, month=1, day=1),
+                datetime.datetime(year=5, month=1, day=1),
+                GREGORIAN,
+                366 * SEC_PER_DAY
+            ),
+            (
+                datetime.datetime(year=4, month=1, day=1),
+                datetime.datetime(year=5, month=1, day=1),
+                NOLEAP,
+                365 * SEC_PER_DAY
+            ),
+            (
+                datetime.datetime(year=300, month=1, day=1),
+                datetime.datetime(year=301, month=1, day=1),
+                GREGORIAN,
+                365 * SEC_PER_DAY
+            ),
+            (
+                datetime.datetime(year=400, month=1, day=1),
+                datetime.datetime(year=400, month=12, day=31),
+                GREGORIAN,
+                365 * SEC_PER_DAY
+            ),
+            (
+                datetime.datetime(year=12, month=7, day=22),
+                datetime.datetime(year=23, month=3, day=15),
+                GREGORIAN,
+                (10 * 365 + 238) * SEC_PER_DAY
+            ),
+            (
+                datetime.datetime(year=1, month=1, day=1),
+                datetime.datetime(year=1001, month=1, day=1),
+                GREGORIAN,
+                (1000 * 365 + 242) * SEC_PER_DAY
+            ),
```

Comment:
These two cases are covered sufficiently by the other tests (unless I missed something)
```suggestion
```

Reply:
I've kept in the first test which was based off Aidan's suggestion to test using some mid-month values, and removed the second one – I agree I think that one's covered by the other cases!